src/garage/_dtypes.py

```diff
@@ -167,3 +167,199 @@ def __new__(cls, env_spec, observations, actions, rewards, terminals,
         return super().__new__(TrajectoryBatch, env_spec, observations,
```

Comment:
you need to add your new types to `garage/__init__.py` so that they can be imported...